File: advanced-programming-methods/toy-interpreter-java/src/model/Pair.java
package model;

/**
 * The type Pair.
 */
public class Pair
{
    private String key;
    private int    value;

    /**
     * Instantiates a new Pair.
     *
     * @param key   the key
     * @param value the value
     */
    public Pair(String key, int value)
    {
        this.value = value;
        this.key = key;
    }

    /**
     * Getter for property 'value'.
     *
     * @return Value for property 'value'.
     */
    public int getValue()
    {
        return value;
    }

    /**
     * Setter for property 'value'.
     *
     * @param value Value to set for property 'value'.
     */
    public void setValue(int value)
    {
        this.value = value;
    }

    /**
     * Getter for property 'key'.
     *
     * @return Value for property 'key'.
     */
    public String getKey()
    {
        return key;
    }

    /**
     * Setter for property 'key'.
     *
     * @param key Value to set for property 'key'.
     */
    public void setKey(String key)
    {
        this.key = key;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String toString()
    {
        return String.format("%s => %d", key, value);
    }
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/Pair.cs
using System;

namespace Model
{
    /// <summary>
    /// The type Pair.
    /// </summary>
    public class Pair
    {
        private string key;
        private int value;

        /// <summary>
        /// Instantiates a new Pair.
        /// </summary>
        /// <param name="key"> the key </param>
        /// <param name="value"> the value </param>
        public Pair(string key, int value)
        {
            this.value = value;
            this.key = key;
        }

        /// <summary>
        /// Getter for property 'value'.
        /// </summary>
        /// <returns> Value for property 'value'. </returns>
        public virtual int Value
        {
            get
            {
                return value;
            }
            set
            {
                this.value = value;
            }
        }


        /// <summary>
        /// Getter for property 'key'.
        /// </summary>
        /// <returns> Value for property 'key'. </returns>
        public virtual string Key
        {
            get
            {
                return key;
            }
            set
            {
                this.key = value;
            }
        }
            
        public override string ToString()
        {
            return string.Format("{0} => {1:D}", key, value);
        }
    }
}

